impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -603,7 +602,15 @@ public float getHitRate(TimeUnit unit) {
             ValueHolder<V> value = fault.get();
             final OnHeapValueHolder<V> newValue;
             if(value != null) {
-              newValue = importValueFromLowerTier(key, value, now);
+              try {
+                newValue = importValueFromLowerTier(key, value, now);
+              } catch (RuntimeException re) {
+                LOG.error("Expiry computation caused an exception - Expiry duration will be 0 ", re);
+                backEnd.remove(key, fault);
+                expireMapping(key, value);
```

Comment:
Unless another thread installs something, this will have no effect as you just removed the mapping ...